python/pyspark/sql/tests.py

```diff
@@ -3654,6 +3654,109 @@ def test_jvm_default_session_already_set(self):
             spark.stop()
 
 
+class SparkSessionTests2(unittest.TestCase):
+
+    def test_active_session(self):
+        spark = SparkSession.builder \
+            .master("local") \
+            .getOrCreate()
+        try:
+            activeSession = SparkSession.getActiveSession()
+            df = activeSession.createDataFrame([(1, 'Alice')], ['age', 'name'])
+            self.assertEqual(df.collect(), [Row(age=1, name=u'Alice')])
+        finally:
+            spark.stop()
+
+    def test_get_active_session_when_no_active_session(self):
+        active = SparkSession.getActiveSession()
+        self.assertEqual(active, None)
+        spark = SparkSession.builder \
+            .master("local") \
+            .getOrCreate()
+        active = SparkSession.getActiveSession()
+        self.assertEqual(active, spark)
+        spark.stop()
+        active = SparkSession.getActiveSession()
+        self.assertEqual(active, None)
```

Comment:
Given the change for how we construct the SparkSession can we add a test that makes sure we do whatever we decide to with the SparkContext?

Reply:
Thanks @holdenk 
I will add a test for the above comment and also add a test for your comment regarding
```
self._jvm.SparkSession.setActiveSession(self._jsparkSession)
```